src/telemetry/basic_robot_human.rs

```diff
@@ -15,76 +15,83 @@ use crate::telemetry::telemetry_structs;
 use crate::telemetry::telemetry_structs::{SnippetTracker, TeleRobotHumanAccum};
 
 
-const ROBOT_HUMAN_FILE_STATS_UPDATE_EVERY: i64 = 15;
+fn update_robot_characters_baseline(
+    rec: &mut TeleRobotHumanAccum,
+    snip: &SnippetTracker
+) {
+    let re = Regex::new(r"\s+").unwrap();
+    let robot_characters = re.replace_all(&snip.grey_text, "").len() as i64;
+    rec.robot_characters_acc_baseline += robot_characters;
+}
 
+fn update_human_characters(
+    rec: &mut TeleRobotHumanAccum,
+    baseline_text: String,
+    text: &String,
+) {
+    let re = Regex::new(r"\s+").unwrap();
+    let (added_characters, removed_characters) = utils::get_add_del_from_texts(&baseline_text, text, true);
+    let real_characters_added = re.replace_all(&added_characters, "").len() as i64 - re.replace_all(&removed_characters, "").len() as i64;
+    let human_characters = real_characters_added - rec.robot_characters_acc_baseline;
+    info!("human_characters: +{}; robot_characters: +{}", human_characters, rec.robot_characters_acc_baseline);
+    rec.human_characters += human_characters;
+    rec.robot_characters += rec.robot_characters_acc_baseline;
+    rec.robot_characters_acc_baseline = 0;
+}
```

Comment:
basetext_to_text_leap_calculations

Reply:
added_lines, removed_lines

Reply:
added_chars, removed_chars = diff(removed_lines, added_lines)
for line in added_lines {
   added_chars_this_line = ...;
   info!("line \"{}\" < human added characters \"{}\", added_chars_this_line);
}